tests/directory_parser/test_directory_parser.py

```diff
@@ -20,14 +20,14 @@ def test_init(self):
         """Testing the basic functionality of the DirectoryParser Class"""
 
         # DirectoryParser returns output
-        parser = DirectoryParser(self.server_address)
+        parser = DirectoryParser(self.wdir)
 
         # relies on the presence of other files in the directory
         # Checks if DirectoryParser lists the server entries
         self.assertNotEqual(parser.entries, [], "parser.entries were not listed")
 
         # path_regex to mozdownload -t release -p win32 -v latest
-        testpath = urljoin(self.wdir, 'directoryparser')
+        testpath = urljoin(self.wdir, 'directoryparser/')
```

Comment:
wptserve will also fail if no final slash has been given for a folder. Maybe @jgraham knows if this is wanted or not.